esp-wifi/src/wifi/mod.rs

```diff
@@ -1629,6 +1625,7 @@ pub(crate) fn wifi_start() -> Result<(), WifiError> {
         cntry_code[2] = crate::CONFIG.country_code_operating_class;
 
         let country = wifi_country_t {
+            #[allow(clippy::useless_transmute)]
```

Comment:
We should probably add a FIXME to remove this after the next rustc release, the c_char redefinition should be stable in 1.85. Needs an MSRV bump for sure, but I'd rather not carry around a transmute when we don't need it.